src/main/java/easton/Easton.java

```diff
@@ -0,0 +1,263 @@
+
+import easton.model.Deadline;
+import easton.model.Event;
+import easton.model.Task;
+import easton.model.ToDo;
+import easton.view.Ui;
+
+import java.io.*;
+import java.util.ArrayList;
+
+public class Easton {
+
+    private ArrayList<Task> tasks = new ArrayList<>();;
+    private Storage storage;
+    private Ui<Task> ui;
+
+    public Easton(String fileName) {
+        try {
+            storage = new Storage(fileName);
+        } catch (IOException e) {
+            Ui.displayText("Cannot connect to the storage.");
+        }
+
+        tasks = retrieveTasks();
+        ui = new Ui<>();
+    }
+
+    public void run() {
+        Ui.welcome();
+        boolean isFinished = false;
+        Action action;
+        String userInput;
+
+        while (!isFinished) {
+            userInput = ui.input();
+            Ui.divider();
+
+            try {
+                action = getActionFromInput(userInput);
+            } catch (IllegalActionException e) {
+                Ui.displayText(e.getMessage());
+                action = Action.INVALID;
+            }
+
+            switch (action) {
+            case BYE:
+                Ui.goodbye();
+                isFinished = true;
+                break;
+            case LIST:
+                ui.list(tasks);
+                break;
+            case MARK:
+                changeTaskStatus(userInput, true, "Nice! I've marked this task as done:");
+                saveTasks();
+                break;
+            case UNMARK:
+                changeTaskStatus(userInput, false, "OK, I've marked this task as not done yet:");
+                saveTasks();
+                break;
+            case TODO:
+                try {
+                    addTask(createToDo(userInput));
+                    saveTasks();
+                } catch (EmptyDescriptionException e) {
+                    Ui.displayText(e.getMessage());
+                }
+                break;
+            case DEADLINE:
+                try {
+                    addTask(createDeadline(userInput));
+                    saveTasks();
+                } catch (EmptyDescriptionException | InvalidFormatException | DateTimeFormatException e) {
+                    Ui.displayText(e.getMessage());
+                }
+                break;
+            case EVENT:
+                try {
+                    addTask(createEvent(userInput));
+                    saveTasks();
+                } catch (EmptyDescriptionException | InvalidFormatException | DateTimeFormatException e) {
+                    Ui.displayText(e.getMessage());
+                }
+                break;
+            case DELETE:
+                deleteTask(userInput);
+                saveTasks();
+                break;
+            }
+
+            Ui.divider();
+        }
+
+    }
+
+    public static void main(String[] args) {
+        new Easton("task.csv").run();
+    }
+
+    private void changeTaskStatus(String input, boolean isDone, String message) {
```

Comment:
Maybe you could move this logic to the Task class?

Reply:
Okay I will look into this!

Reply:
@weijianwong looking into the logic of this function, changeTaskStatus(). I'm worried if I shifted the logic into the Task class, specfically the printing of statement, that I would create a dependency assiocation between Task and the Ui classes which I forsee the code becoming more buggy if one of the parts were to break. But, yes input is appreciated.